### DISCUSSION TITLE
Interact with charts

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#6364] [Ideas]
Charts in BB are beautiful and very easy to set up.
It would be very powerful if we could interact with them : trigger an action on click on a chart area : open a screen, run a query, replace with a new chart (to simulate drilldown) ...

**Comment by melohagan** [comment]
Cool idea! 

I think this would be dependant on the flexibility of the chart library that's being used in Budibase.

You can obviously use containers and put buttons beside your charts, but that wouldn't cover the "drill into" type functionality for your data.

**Comment by NicolasFabre81** [comment]
With these buttons, would I be able to customize the chart itself ? For
example change the numeric variable that is currently in use in the chart...

**Reply by melohagan** [reply]
Ah good point - most of the settings don't have bindings, so this would probably be the first iteration of this feature if it was to be built.

**Comment by benaunt** [comment]
Is there any updates on this.

It would be useful to add on click actions to the data in charts eg clicking a piece of the pie chart to:

- Dynamically filter the page
- Navigate to another page and pre-filter a table based on the data clicked on etc.
